diff_drive_controller/test/diff_drive_test.cpp

```diff
@@ -54,11 +52,15 @@ TEST_F(DiffDriveControllerTest, testForward)
 
   nav_msgs::Odometry new_odom = getLastOdom();
 
+  const ros::Duration actual_elapsed_time = new_odom.header.stamp - old_odom.header.stamp;
+
+  const double expected_distance = cmd_vel.linear.x * actual_elapsed_time.toSec();
+
   // check if the robot traveled 1 meter in XY plane, changes in z should be ~~0
   const double dx = new_odom.pose.pose.position.x - old_odom.pose.pose.position.x;
   const double dy = new_odom.pose.pose.position.y - old_odom.pose.pose.position.y;
   const double dz = new_odom.pose.pose.position.z - old_odom.pose.pose.position.z;
-  EXPECT_NEAR(sqrt(dx*dx + dy*dy), 1.0, POSITION_TOLERANCE);
+  EXPECT_NEAR(sqrt(dx*dx + dy*dy), expected_distance, POSITION_TOLERANCE);
```

Comment:
@bmagyar here is the proposed change.

Reply:
awesome!

Reply:
Not really... Now [l.74](https://github.com/ros-controls/ros_controllers/pull/318/files/747ae347a7b06bfa2b8ffce6852c7e392a4102a1..fdcd29338656cb89e4d80c66c75326b492a8aff4#diff-3e80dfcdac2700fef577d7a8033ff8aaR74) fails (sometime)...  
This quite looks like a clock issue. I will try later having the `diffbot` publishing it's own clock and `use_sim_time true`.